src/transformers/models/t5/modeling_t5.py

```diff
@@ -808,16 +808,16 @@ def forward(
         
         elif self.memory_efficient_attention in ["torch_scaled_dot_product_attention", "torch_scaled_dot_product_attention_force_flash", "torch_scaled_dot_product_attention_without_position_bias"]:
             scale = 1 / math.sqrt(K) if self.scale == None else self.scale
-            is_causal = self.is_decoder and not self.cross_attention
             if self.memory_efficient_attention in ["torch_scaled_dot_product_attention_force_flash", "torch_scaled_dot_product_attention_without_position_bias"]:
+                is_causal = self.is_decoder and not self.cross_attention
                 with torch.backends.cuda.sdp_kernel(
                     enable_flash=True, 
                     enable_math=self.memory_efficient_attention != "torch_scaled_dot_product_attention_force_flash",
                     enable_mem_efficient=self.memory_efficient_attention != "torch_scaled_dot_product_attention_force_flash",
                 ):
                     attn_output = F.scaled_dot_product_attention(query_states, key_states, value_states, attn_mask=None, dropout_p=self.dropout, is_causal=is_causal, scale=scale)
             else:
-                attn_output = F.scaled_dot_product_attention(query_states, key_states, value_states, attn_mask=add_to_scores, dropout_p=self.dropout, is_causal=is_causal, scale=scale)
+                attn_output = F.scaled_dot_product_attention(query_states, key_states, value_states, attn_mask=add_to_scores, dropout_p=self.dropout, is_causal=False, scale=scale)
```

Comment:
how do you know it's not casual here ?

Reply:
The attn_mask makes it causal or not.
I got an error when specified both attn_mask and is_casual.